File: backend/app/schemas/transactions.py
"""Transaction analysis schemas."""
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel, Field


class ContractSummaryItem(BaseModel):
    entity: str
    total_ar: float = 0
    total_ap: float = 0
    net_exposure: float = 0
    period: str


class OrderSummaryItem(BaseModel):
    period: str
    revenue: float = 0
    cost: float = 0
    profit: float = 0


class ProjectSummaryItem(BaseModel):
    entity: str
    total_revenue: float = 0
    total_cost: float = 0
    profit_margin: float = 0
    period_span: str = ""


class AnomalyItem(BaseModel):
    metric_name: str
    period: str
    value: float = 0
    expected_mean: float = 0
    sigma_distance: float = 0
    entity: Optional[str] = None


class LargeAmountItem(BaseModel):
    metric_name: str
    metric_value: float = 0
    period: str
    entity: Optional[str] = None


class TransactionQueryParams(BaseModel):
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=200)
    period: Optional[str] = None
    entity: Optional[str] = None
    period_from: Optional[str] = None
    period_to: Optional[str] = None
    min_value: Optional[float] = None
    max_value: Optional[float] = None
    threshold: float = Field(default=1000000, ge=0)
    metric_names: Optional[str] = None

File: frontend/src/api/transactions.ts
import { get } from './request'

export interface ContractItem { entity: string; total_ar: number; total_ap: number; net_exposure: number; period: string }
export interface OrderItem { period: string; revenue: number; cost: number; profit: number }
export interface ProjectItem { entity: string; total_revenue: number; total_cost: number; profit_margin: number; period_span: string }
export interface AnomalyItem { metric_name: string; period: string; value: number; expected_mean: number; sigma_distance: number; entity: string | null }
export interface LargeAmountItem { metric_name: string; metric_value: number; period: string; entity: string | null }

export function getContracts(params?: Record<string, unknown>) { return get('/transactions/contracts', { params }) }
export function getOrders(params?: Record<string, unknown>) { return get('/transactions/orders', { params }) }
export function getProjects(params?: Record<string, unknown>) { return get('/transactions/projects', { params }) }
export function getAnomalies(params?: Record<string, unknown>) { return get('/transactions/anomalies', { params }) }
export function getLargeAmounts(params?: Record<string, unknown>) { return get('/transactions/large-amounts', { params }) }
